catalog/views.py

```diff
@@ -0,0 +1,203 @@
+from django.contrib.auth import logout
+from django.contrib.auth.decorators import login_required
+from django.contrib.auth.mixins import LoginRequiredMixin
+from django.http import HttpResponseRedirect
+from django.shortcuts import redirect
+from django.shortcuts import render, get_object_or_404
+from django.urls import reverse_lazy
+from django.views.generic import (
+    ListView,
+    DetailView,
+    UpdateView,
+    CreateView,
+    DeleteView,
+)
+
+from catalog.forms import (
+    CookSearchForm,
+    DishSearchForm,
+    DishTypeSearchForm,
+    CookExperienceUpdateForm,
+    DishForm,
+)
+from catalog.models import Cook, Dish, DishType
+
+
+@login_required
+def index(request):
+    """View function for the home page of the site."""
+
+    num_cooks = Cook.objects.only("id").count()
+    num_dishes = Dish.objects.only("id").count()
+    num_dish_types = DishType.objects.only("id").count()
+
+    num_visits = request.session.get("num_visits", 0)
+    request.session["num_visits"] = num_visits + 1
+
+    context = {
+        "num_cooks": num_cooks,
+        "num_dishes": num_dishes,
+        "num_dish_types": num_dish_types,
+        "num_visits": num_visits + 1,
+    }
+
+    return render(request, "catalog/index.html", context=context)
+
+
+class CookListView(LoginRequiredMixin, ListView):
```

Comment:
Dla ujednolicenia. Na górze używasz dekoratora, więc tu możemy pójść w tą samą stronę dla czytelności
```suggestion
from django.utils.decorators import method_decorator

@method_decorator(login_required, name='dispatch')
class CookListView(ListView):
```